src/Uno.UI/Microsoft/UI/Xaml/Controls/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	// UNO Docs:
+	// In WinUI 2.6. RadioMenuFlyoutItem derives publically from MenuFlyoutItem, but secretly from ToggleMenuFlyoutItem.
+	// Since we can't do that in C#, the important functionality are in ToggleMenuFlyoutItem, we derive from it.
+	public partial class RadioMenuFlyoutItem : ToggleMenuFlyoutItem
+	{
+		[ThreadStatic]
+		private static Dictionary<string, WeakReference<RadioMenuFlyoutItem>> s_selectionMap;
+
+		// Copies of IsChecked & GroupName to avoid using those dependency properties in the ~RadioMenuFlyoutItem() destructor which would lead to crashes.
+		private bool m_isChecked;
+		private string m_groupName;
+
+		private bool m_isSafeUncheck;
+
+		public RadioMenuFlyoutItem()
+		{
+			this.RegisterPropertyChangedCallback(ToggleMenuFlyoutItem.IsCheckedProperty, OnInternalIsCheckedChanged);
+
+
+			if (s_selectionMap == null)
+			{
+				// Ensure that this object exists
+				s_selectionMap = new Dictionary<string, WeakReference<RadioMenuFlyoutItem>>();
+			}
+
+			SetDefaultStyleKey(this);
+
+		}
+
+		~RadioMenuFlyoutItem()
+		{
+			// If this is the checked item, remove it from the lookup.
+			if (m_isChecked)
+			{
+				if (m_groupName != null)
+				{
+					s_selectionMap.Remove(m_groupName);
```

Comment:
This is likely to fail because the finalizer is called from another thread. We need to alter this by locking around `s_selectionMap`, or used a thread safe collection.